### DISCUSSION TITLE
How to configure `pyright` to respect `pyproject.toml`

### DISCUSSION THREAD

**Discussion by dycw** [williamboman/nvim-lsp-installer#569] [Q&A]
I am using `nvim-lsp-installer` through [`LunarVim`](https://github.com/LunarVim/LunarVim). I love how everything's setup, but now I'm starting to want to learn to configure `pyright` to respect `pyproject.toml`.

![Screenshot from 2022-04-02 16-22-34](https://user-images.githubusercontent.com/10572122/161372036-771d8d23-4951-424c-8ba4-9faa476ec86a.png)

In the screenshot above:

- Top pane: `pyright` through `neovim` is reporting a problem.
- Middle pane: `pyright` through the CLI is not.
- Bottom pane: The expected behaviour is to see no problem; `reportUnknownArgumentType` is disabled.

Any advice please? I've seen #304 but I couldn't seem to get help from it.

**Comment by williamboman** [comment]
Hey! Have you seen the [configuration docs for pyright](https://github.com/microsoft/pyright/blob/main/docs/configuration.md)? They mention the following:

> Pyright settings can also be specified in a [tool.pyright] section of a “pyproject.toml” file. A “pyrightconfig.json” file always takes precedent over “pyproject.toml” if both are present.

In your `pyproject.toml`, are your settings grouped inside a `[tool.pyright]` section?

**Reply by dycw** [reply]
Hi @williamboman , thanks for the reply.

For me, CLI `pyright` is

```bash
❯ which pyright
/home/linuxbrew/.linuxbrew/bin/pyright

❯ pyright --version
pyright 1.1.234
```

and this I am in configuring using a `[tool.pyright]` section. The screenshot above shows my setting for ``reportUnknownArgumentType`` as being disabled, and the middle pane show it's correctly disabled — CLI `pyright` reports no errors.

However, `neovim`'s LSP `pyright` (which I'm guessing is different from my `linuxbrew`-installed `pyright`) is reporting my code as having an error (top screenshot). This seems to not be respecting my project's `[tool.pyright]` section.
